### ISSUE TITLE
6.0.2 support

### ISSUE THREAD

**Issue by crazy-max** [dscharrer/innoextract#83]
Hi,

While trying to extract the official setup [innosetup-6.0.2.exe](http://files.jrsoftware.org/is/6/innosetup-6.0.2.exe) :

```
Warning: Unexpected setup data version: 6.0.0 (unicode)
Warning: Unexpected Auto Boolean value: 10
Stream error while parsing setup headers!
 ├─ detected setup version: 6.0.0 (unicode)
 └─ error reason: basic_ios::clear
If you are sure the setup file is not corrupted, consider
filing a bug report at http://innoextract.constexpr.org/issues
Done with 1 error and 2 warnings.
```

**Comment by bovirus** [comment]
@dscharrer 

Could you pelase compile binaries (for Windwos and Linux) that support Inno Setup 6.0.2

**Comment by PrimordialHelios** [comment]
I can confirm that this affects many games on GOG, since they all use newer innosetup versions.

**Comment by bovirus** [comment]
@dscharrer 

What's about the new binary file to decompress Inno Setup 6..x. installer file?

**Comment by bovirus** [comment]
@dscharrer 

I compiled latest build (1.8 based on June 2019 souyrce code) under Ubuntu but is working only under Linux (not Windows).
Is it possible under Linux to build also the .exe file for Windows?

The snapshot files stopped at January 2019.
And the latest snapshot support only InnoS etup up to 5.6.1

chocolate build is not update (1.6).

**Comment by bovirus** [comment]
@dscharrer 

Some could be make a compialtion of package .8 dev for Windows

Why is not available the 1.8 dev for Linux as apckahe?

**Comment by bovirus** [comment]
@dscharrer

Why you 

didn't answer
didn't updateyour web site
didn't relòase a beta build

with support of Inno Setup 6.0.2?

**Comment by vv221** [comment]
> Why you
> 
> didn't answer

He probably has a life.

> didn't updateyour web site

Website is up to date, latest release being 1.7.

> didn't relòase a beta build
> 
> with support of Inno Setup 6.0.2?

You can already build innoextract with support for InnoSetup 6.0.2 by taking any commit from https://github.com/dscharrer/innoextract/commit/5205b02b272ac1540b58eb6a05df550f214c8fa4
Building current master would include InnoSetup 6.0.2 support.

**Comment by bovirus** [comment]
I believe that if he can have time to update repository can have time to build a beta for linux and windows.
Otherwise no obe can test ir.
The latest snapshot is of 2019.02 without support of Inno Setup 6.0.2.
In the web site there is no mention about new  beta build with support at inno setup 6.0.2.
I need windows version with support for 6.0.2.

**Comment by bovirus** [comment]
@rewolff  

Are you able to compile a Windows version  of innoextract using latest source code in github to enable unapck of Inno Setup 6.0.2 installer in Windows?

Thanks.


@dscharrer 

Could you pelase make a Windows build of innoextract using latest source code and post this temporary build here

https://constexpr.org/innoextract/files/snapshots/

Thanks.

**Comment by rewolff** [comment]
> @rewolff Are you able to compile a Windows version 
No. 

I get a headache trying to use windows. In fact getting one now. Provided a source tree to someone. Turns out windows unzip overwrites somefile.S with the contents of somefile.s, when the latter is in the zip after the first.  (the .s was, it seems, an accident that happened three years ago, but has not bothered me all those years).

**Comment by MasterO2** [comment]
In all fairness to bovirus, he has a point.  Support for Inno Setup 6.0.2 installers is really important, since many GOG installers now use this version.  It's not like anyone's asking you guys to get to the moon...

**Comment by bovirus** [comment]
@MasterO2 

Thanks for the confirmation.

For me it's quite frustrating have the source update but binaries not compiled.
I beleieve that for me and most the user the important resources are the binary file.

I didn't understand what's the problem for the author to compile the Windows binaries.

And with the binaries for Windows and Linux, if uploaded on web page, can enable the users to use these new binareis and reoprt if it works fine or not.

**Comment by vv221** [comment]
> Support for Inno Setup 6.0.2 installers is really important, since many GOG installers now use this version.

Currently GOG installers seem to use a non standard 5.6.2 version, not the 6.0.2 one.
There is already another issue about 5.6.2 support: #74 

Anyway, current master already supports both. Someone wanting to use this development version only needs to build it from source, or rent someone services to do it for them.
**Demanding** (≠ "asking") free service on this front is not going to be well-received.

**Comment by bovirus** [comment]
@vv221 

Time by time many application with an installer based on Inno Setup are moving from vesrion 5.xx to version 6.xx of Inno Setup.

Then current innoectract version is coming time by time less useful because cannot open new exe file based on Inno Setup 6.xx.

Please takle care that other Inno Setup tools (ex. innounp) has already released binaries for Windows and Linux compatible with Inno Setup 6.xx exe file.

Then, if you consider that the innoextract source code is already done for Inno Setup 6.xx I don't undesrstand because the author didn't build binaries based on latest source code.

And of course because he didn't answer explaining his position.

**Comment by vv221** [comment]
> I posted here #93 a request but no one (oinlcuded the source developer) answer.

You posted an issue two days ago, and are complaining because it doesn’t already have an answer by the developer. That’s because of this kind of behaviour that your posts get ignored.
And that’s a shame, because barring that the issue you opened is pertinent and could bring benefits to Windows users.

By the way, from now on I’m going to do the same, because it does not seem it’s worth the time I’m spending on your posts. Feel free to **not** notify me again.

**Comment by bovirus** [comment]
@W221

My request is two days ago but the problem about it is more or less is the same from long time (January 2019).

The source code is updated to supprt Inno Setup 6,.0.2.  

But is not available Linux/Windows binarie based on laetst source cdoe,.

After a lot of requests where no one answered from long time (sofwtare author included), I decide to try myself to compile Windows binaries.

But after that I check that the author didn't post any info how to compile binareis for Windows.

I beleive that the info about 

source code
compilation instructions
website information

are important and are old (not updated for a long time) and it reduce the option to use this program.

Always considering that other tools (like innounp) are already updated to support fully Inno Setup 6.0.2 exe installer file.

**Comment by bovirus** [comment]
My big misundersatnding is because the author

didn't answer/post in this repository (just update source code).
didn't make binaries for Windows
didn't make binaries for Linux

**Comment by dscharrer** [comment]
innoextract 1.8 has been released with 6.0.2 support.
